File: gitdot-web/packages/api/src/endpoint/organization/update-organization.ts
import { z } from "zod";
import { OrganizationResource } from "../../resource";
import type { Endpoint } from "../endpoint";

export const UpdateOrganizationRequest = z.object({
  location: z.string().nullable().optional(),
  readme: z.string().nullable().optional(),
  links: z.array(z.string()).optional(),
});
export type UpdateOrganizationRequest = z.infer<
  typeof UpdateOrganizationRequest
>;

export const UpdateOrganizationResponse = OrganizationResource;
export type UpdateOrganizationResponse = z.infer<
  typeof UpdateOrganizationResponse
>;

export const UpdateOrganization = {
  path: "/organization/{org_name}",
  method: "PATCH",
  request: UpdateOrganizationRequest,
  response: UpdateOrganizationResponse,
} as const satisfies Endpoint;
export type UpdateOrganization = typeof UpdateOrganization;

File: gitdot-api/src/endpoint/organization/update_organization.rs
use serde::{Deserialize, Serialize};

use crate::{endpoint::Endpoint, resource::OrganizationResource};

pub struct UpdateOrganization;

impl Endpoint for UpdateOrganization {
    const PATH: &'static str = "/organization/{org_name}";
    const METHOD: http::Method = http::Method::PATCH;

    type Request = UpdateOrganizationRequest;
    type Response = UpdateOrganizationResponse;
}

#[derive(ApiRequest, Debug, Serialize, Deserialize)]
pub struct UpdateOrganizationRequest {
    pub location: Option<String>,
    pub readme: Option<String>,
    pub links: Option<Vec<String>>,
}

pub type UpdateOrganizationResponse = OrganizationResource;
